diff -Naur glm-0.9.3.2/glm/core/type_vec1.hpp glm-0.9.3.2-patch/glm/core/type_vec1.hpp
--- glm-0.9.3.2/glm/core/type_vec1.hpp	2012-03-16 21:56:52.000000000 -0700
+++ glm-0.9.3.2-patch/glm/core/type_vec1.hpp	2012-04-21 21:58:21.737484530 -0700
@@ -35,6 +35,8 @@
 #include "type_size.hpp"
 #include "_swizzle.hpp"
 
+#include "msgpack.hpp"
+
 namespace glm{
 namespace detail
 {
@@ -67,6 +69,8 @@
 #		else//(GLM_COMPONENT == GLM_COMPONENT_GLSL_NAMES)
 			union {value_type x, r, s;};
 #		endif//GLM_COMPONENT
+            
+        MSGPACK_DEFINE(x);
 
 		//////////////////////////////////////
 		// Accesses
diff -Naur glm-0.9.3.2/glm/core/type_vec2.hpp glm-0.9.3.2-patch/glm/core/type_vec2.hpp
--- glm-0.9.3.2/glm/core/type_vec2.hpp	2012-03-16 21:56:52.000000000 -0700
+++ glm-0.9.3.2-patch/glm/core/type_vec2.hpp	2012-04-21 21:58:59.744150449 -0700
@@ -35,6 +35,8 @@
 #include "type_size.hpp"
 #include "_swizzle.hpp"
 
+#include "msgpack.hpp"
+
 namespace glm{
 namespace detail
 {
@@ -98,6 +100,8 @@
 			GLM_SWIZZLE_GEN_VEC_FROM_VEC2_COMP(value_type, detail::tvec2, detail::tvec2, detail::tvec3, detail::tvec4, x, y)
 #		endif//(defined(GLM_SWIZZLE))
 #	endif//GLM_COMPONENT
+            
+        MSGPACK_DEFINE(x, y);
 
 		//////////////////////////////////////
 		// Accesses
diff -Naur glm-0.9.3.2/glm/core/type_vec3.hpp glm-0.9.3.2-patch/glm/core/type_vec3.hpp
--- glm-0.9.3.2/glm/core/type_vec3.hpp	2012-03-16 21:56:52.000000000 -0700
+++ glm-0.9.3.2-patch/glm/core/type_vec3.hpp	2012-04-21 21:59:52.017482759 -0700
@@ -35,6 +35,8 @@
 #include "type_size.hpp"
 #include "_swizzle.hpp"
 
+#include "msgpack.hpp"
+
 namespace glm{
 namespace detail
 {
@@ -99,6 +101,8 @@
 			GLM_SWIZZLE_GEN_VEC_FROM_VEC3_COMP(T, detail::tvec3, detail::tvec2, detail::tvec3, detail::tvec4, x, y, z)
 #		endif//(defined(GLM_SWIZZLE))
 #	endif//GLM_COMPONENT
+            
+        MSGPACK_DEFINE(x, y, z);
 
 		//////////////////////////////////////
 		// Accesses
diff -Naur glm-0.9.3.2/glm/core/type_vec4.hpp glm-0.9.3.2-patch/glm/core/type_vec4.hpp
--- glm-0.9.3.2/glm/core/type_vec4.hpp	2012-03-16 21:56:52.000000000 -0700
+++ glm-0.9.3.2-patch/glm/core/type_vec4.hpp	2012-04-21 22:00:18.510815572 -0700
@@ -35,6 +35,8 @@
 #include "type_size.hpp"
 #include "_swizzle.hpp"
 
+#include "msgpack.hpp"
+
 namespace glm{
 namespace detail
 {
@@ -100,6 +102,8 @@
 			GLM_SWIZZLE_GEN_VEC_FROM_VEC4_COMP(T, detail::tvec4, detail::tvec2, detail::tvec3, detail::tvec4, x, y, z, w)
 #		endif//(defined(GLM_SWIZZLE))
 #	endif//GLM_COMPONENT
+            
+        MSGPACK_DEFINE(x, y, z, w);
 
 		//////////////////////////////////////
 		// Accesses
