diff -Naur libuv-0.6/src/unix/fs.c libuv-0.6-patch/src/unix/fs.c
--- libuv-0.6/src/unix/fs.c	2012-03-29 19:24:37.000000000 -0700
+++ libuv-0.6-patch/src/unix/fs.c	2012-04-21 20:59:17.390887360 -0700
@@ -692,7 +692,7 @@
   req->work_cb = work_cb;
   req->after_work_cb = after_work_cb;
 
-  req->eio = eio_custom(uv__work, EIO_PRI_DEFAULT, uv__after_work, req);
+  req->eio = eio_custom(uv__work, EIO_PRI_MAX, uv__after_work, req);
 
   if (!req->eio) {
     uv__set_sys_error(loop, ENOMEM);
