diff -Naur LuaJIT-2.0.0/src/Makefile LuaJIT-2.0.0-patch/src/Makefile
--- LuaJIT-2.0.0/src/Makefile	2012-11-08 02:10:00.000000000 -0800
+++ LuaJIT-2.0.0-patch/src/Makefile	2013-02-07 22:15:04.809910332 -0800
@@ -49,7 +49,7 @@
 # the binaries to a different machine you could also use: -march=native
 #
 CCOPT_x86= -march=i686
-CCOPT_x64=
+CCOPT_x64= -march=native
 CCOPT_arm=
 CCOPT_ppc=
 CCOPT_ppcspe=
@@ -72,10 +72,10 @@
 # as dynamic mode.
 #
 # Mixed mode creates a static + dynamic library and a statically linked luajit.
-BUILDMODE= mixed
+#BUILDMODE= mixed
 #
 # Static mode creates a static library and a statically linked luajit.
-#BUILDMODE= static
+BUILDMODE= static
 #
 # Dynamic mode creates a dynamic library and a dynamically linked luajit.
 # Note: this executable will only run when the library is installed!
